aten/src/THC/THCTensorRandom.cu

```diff
@@ -62,19 +64,32 @@ __global__ void set_rngstate_kernel(curandStateMtgp32 *state, mtgp32_kernel_para
 
 __host__ void THCRandom_setRNGState(THCState* state, THByteTensor *rng_state)
 {
-  Generator* gen = THCRandom_getGenerator(state);
+  THCGenerator* gen = THCRandom_getGenerator(state);
 
   static const size_t states_size = MAX_NUM_BLOCKS * sizeof(curandStateMtgp32);
   static const size_t seed_size = sizeof(gen->initial_seed);
-  static const size_t total_size = states_size + seed_size;
-  THArgCheck(THByteTensor_nElement(rng_state) == total_size, 1, "RNG state is wrong size");
+  static const size_t offset_size = sizeof(gen->philox_seed_offset);
+  static const size_t total_size = states_size + seed_size + offset_size;
+  bool deprecated = false;
+  if (THByteTensor_nElement(rng_state) == total_size - offset_size) {
+    deprecated = true;
+  }
+  else {
+    THArgCheck(THByteTensor_nElement(rng_state) == total_size, 1, "RNG state is wrong size");
```

Comment:
Doesn't it fail if you try to set it now? You're not checking for the case where we don't trigger the old behavior.

Reply:
I think this case checks it? Unless I'm missing something - if the size of `rng_state` is `total_size = states_size + seed_size + offset_size`, it'll work along this code path, otherwise fail. 

It might be more clear as `else if (...) { THArgError("RNG state is wrong size"); }` where `...` is the condition checked on line 78.

Reply:
Ah, my bad. Never mind the comment